src/cmdstan/command_helper.hpp

```diff
@@ -767,22 +768,40 @@ void init_callbacks(
         &diag_csv_writers,
     std::vector<stan::callbacks::json_writer<std::ofstream>>
         &diag_json_writers) {
+  // bookkeeping
   auto user_method = parser.arg("method");
   unsigned int num_chains = get_num_chains(parser);
   unsigned int id = get_arg_val<int_argument>(parser, "id");
   int sig_figs = get_arg_val<int_argument>(parser, "output", "sig_figs");
+  bool is_pathfinder = user_method->arg("pathfinder");
+  bool save_single_paths
+      = is_pathfinder && num_chains > 1
+        && get_arg_val<bool_argument>(parser, "method", "pathfinder",
+                                      "save_single_paths");
 
+  // output.csv files for all chains and single-path-pathfinder csv files
   sample_writers.reserve(num_chains);
-  std::vector<std::string> output_filenames
-      = make_filenames(get_arg_val<string_argument>(parser, "output", "file"),
-                       ".csv", num_chains, id);
-  for (int i = 0; i < num_chains; ++i) {
-    auto ofs = std::make_unique<std::ofstream>(output_filenames[i]);
-    if (sig_figs > -1)
-      ofs->precision(sig_figs);
-    sample_writers.emplace_back(std::move(ofs), "# ");
+  if (!is_pathfinder || save_single_paths || num_chains == 1) {
+    std::vector<std::string> output_filenames
+        = make_filenames(get_arg_val<string_argument>(parser, "output", "file"),
+                         "", ".csv", num_chains, id);
+    if (save_single_paths) {
+      output_filenames = make_filenames(
+          get_arg_val<string_argument>(parser, "output", "file"), "_path",
+          ".csv", num_chains, id);
+    }
+    for (int i = 0; i < num_chains; ++i) {
+      auto ofs = std::make_unique<std::ofstream>(output_filenames[i]);
+      if (sig_figs > -1)
+        ofs->precision(sig_figs);
+      sample_writers.emplace_back(std::move(ofs), "# ");
+    }
+  } else if (is_pathfinder && !save_single_paths) {
+    for (int i = 0; i < num_chains; ++i) {
+      sample_writers.emplace_back(nullptr, "# ");
+    }
```

Comment:
I'm happy to be overruled on this but I think the intention reads clearer if this special case is an if, and then everything else is just an 'else'

So you'd have (I think) `  if (is_pathfinder && !(save_single_paths && num_chains > 1) ) {` followed by this block, and then an `else` with all the above.

A comment explaining why we're branching would also be nice!